source/bayes_simulation.Rmd

```diff
@@ -45,112 +44,268 @@ correctly identifies 80 percent as "OK" but says that 20 percent are
+1. Note that we are only interested in outcomes where the mechanic
+   approved a car.
 
-The following steps determine the probability that a car *said*
+2. We start with N cars.  For each car, generate a label of either
+   "faulty" or "working" with probabilities of 0.3 and 0.7, respectively.
 
-to be "OK" will turn out to be *really* faulty:
+3. For each faulty car, we generate one of two labels, "approved" or
+   "not approved" with probabilities 0.1 and 0.9, respectively.
 
-1.  Model in percentages the universe of all cars as a bucket of 100
-    balls. Working from the data as given above, and referring to first
-    principles, color (.9 \* .3 =) .27 of the 100 balls violet (the 27
-    faulty balls said to be "faulty"), (.1 \* .3 =) .03 of the 100 balls
-    blue (3 balls said "OK" but faulty), (.2 \* .7 =) .14 of the balls
-    orange (14 OK cars said to be "faulty" balls), and (.8 \* .7
+4. For each working car, we generate one of two labels, "approved" or
+   "not approved" with probabilities 0.7 and 0.3, respectively.
 
-    = ) 56 balls maroon (said to be "OK" that really are OK). A Venn
-    diagram may help with this step, but it is not necessary.
+5. Out of all cars "approved", count how many are "faulty".
+   The ratio between these numbers is our answer.
 
-    An even better procedure would be to work directly from the concrete
-    data. One would note, for example, that of 200 cars previously
-    observed, 54 were faulty and were said to be "faulty," 6 were faulty
-    and were said to be "OK," 28 were OK but were said to be "faulty,"
-    and 112 were OK and were said to be "OK." Then make a bucket of 54
-    violets, 6 blue, 28 orange, and 112 maroon.
+Here is the whole thing:
 
-2.  Draw a ball. If it is one of those said to be "faulty"---that is,
-    violet or orange---draw (with replacement) another ball. Continue
-    until obtaining a ball said to be "OK"---that is, a blue or maroon
-    ball. Then record its color.
+```{python}
+import numpy as np
 
-3.  Repeat step 2 perhaps 1000 times and compute the proportions of
-    blues among the recorded results.
+N = 1000000  # number of cars
 
--OR-
+# Generate an array with as many entries as there are cars, each
+# being either 'working' or 'faulty'
+cars = np.random.choice(['working', 'faulty'], p=[0.7, 0.3], size=N)
 
-1.  Choose a number randomly between "1" and "100."
+# Count how many cars are working and how many are faulty
+N_working = np.count_nonzero(cars == 'working')
+N_faulty = N - N_working
 
-2.  If "28-30" or "31-45:, record; otherwise draw another number, and
-    repeat until a number is recorded.
+# Create a new array in which to stores what a mechanic says
+# about the car: 'approved' or 'not approved'
+mechanic_says = np.empty_like(cars, dtype=object)
 
-3.  Repeat step 2 and count the proportion "28-30" among the total
-    "28-45."
+# For each working car, generate a mechanic mechanic_says/declared faulty
```

Comment:
Wasn't quite sure what you meant.  Maybe "mechanic_says" value, either "approved" or "not approved"?

Reply:
No idea what happened here.